### PR TITLE
Added the ability to show the number of tweet replies

### PR DESCRIPTION
**Pull request by robertmrobo**
This is a PR to an Issue I opened yesterday. #99

Screenshot 2022-04-08 at 20 10 44

### CODE REVIEW

File: lib/embedded_tweet_view.dart [hunk 1]
```diff
@@ -25,6 +25,9 @@ class EmbeddedTweetView extends StatelessWidget {
   /// If set to true the the text and icons will be light
   final bool darkMode;
 
+  /// If set to true,
```
Comment (wszeborowskimateusz) [inline lib/embedded_tweet_view.dart:28]: I think it misses the end part?

File: lib/models/viewmodels/converters/tweet_v2_to_tweet_vm_converter.dart [hunk 2]
```diff
@@ -220,10 +222,12 @@ class TweetV2ToTweetVMConverter {
   }
 
   int get _favoriteCount => tweet.publicMetrics.likeCount;
+  int get _repliesCount => tweet.publicMetrics.replyCount;
```
Comment (wszeborowskimateusz) [inline lib/models/viewmodels/converters/tweet_v2_to_tweet_vm_converter.dart:225]: I think for consistency you can add line break between _favoriteCount and _repliesCount

File: lib/models/viewmodels/converters/tweet_v2_to_tweet_vm_converter.dart [hunk 3]
```diff
@@ -220,10 +222,12 @@ class TweetV2ToTweetVMConverter {
...
 
   int get _startDisplayText => 0;
 
   int _endDisplayText(TweetV2 tweet) => tweet.text.length;
 
   bool get _favorited => _favoriteCount > 0;
+  bool get _replied => _repliesCount > 0;
```
Comment (wszeborowskimateusz) [inline lib/models/viewmodels/converters/tweet_v2_to_tweet_vm_converter.dart:232]: As above - line break

File: lib/embedded_tweet_view.dart [hunk 4]
```diff
@@ -46,6 +49,7 @@ class EmbeddedTweetView extends StatelessWidget {
     this.onTapImage,
     this.createdDateDisplayFormat,
     required this.videoHighQuality,
+    this.showRepliesCount = false,
```
Comment (wszeborowskimateusz) [inline lib/embedded_tweet_view.dart:52]: We just add it to embedded tweet view?

**Comment by wszeborowskimateusz** [comment]
Could you add some example image to the PR?

File: lib/embedded_tweet_view.dart [hunk 5]
```diff
@@ -207,6 +213,25 @@ class EmbeddedTweetView extends StatelessWidget {
                             color: (darkMode)
                                 ? Colors.grey[400]
                                 : Colors.grey[600]))),
+                if (showRepliesCount && _tweetVM.repliesCount != null)
```
Comment (wszeborowskimateusz) [inline lib/embedded_tweet_view.dart:216]: I think this can be extracted to some getter and reused below

**Comment by wszeborowskimateusz** [comment]
Also please Update the CHANGELOG file (Unreleased part)

And please add yourself to contributors section of README file :)

**Comment by robertmrobo** [comment]
> Could you add some example image to the PR?

Just did, please bear with me, this is the first-ever OSS contribution I've ever done in my life. But I'm glad you've been kind and warm so far as I make my first baby steps.

File: lib/embedded_tweet_view.dart (hunk 5 above)
Comment (robertmrobo) [reply lib/embedded_tweet_view.dart:216]: What do you think about the following?

I'm thinking of adding the following method on the VM.

`bool shouldShowReplies(bool shouldShow) => shouldShow && repliesCount != null; //I'll come up with a better param name if we are to ahead with this :)`

and then on the tweet view we'll do:

`if (_tweetVM.shouldShowReplies(showRepliesCount)){
       //show replies
}`

File: lib/embedded_tweet_view.dart (hunk 4 above)
Comment (robertmrobo) [reply lib/embedded_tweet_view.dart:52]: Well, from the examples, I only see the public metrics only displayed on the Embedded tweets. And when I tried to look at the Tweet and Compact views, I can't get clearly where to add this 🙈.

I'm a noob as far as frontend is concerned. Been a backend developer my whole life. Now I'm just converting to full-stack

Screenshot 2022-04-08 at 23 54 26
.

File: lib/embedded_tweet_view.dart (hunk 5 above)
Comment (wszeborowskimateusz) [reply lib/embedded_tweet_view.dart:216]: I was more thinking about a getter in EmbeddedTweetView

```bool get _shouldShowReplies => showRepliesCount && _tweetVM.repliesCount != null```

File: lib/embedded_tweet_view.dart (hunk 4 above)
Comment (wszeborowskimateusz) [reply lib/embedded_tweet_view.dart:52]: ok, then let's leave it like this.

File: lib/embedded_tweet_view.dart (hunk 5 above)
Comment (robertmrobo) [reply lib/embedded_tweet_view.dart:216]: Ok, that's fine, I thought maybe what I proposed above will be equally reusable should the need arise to add this feature to other views as well. Compact and Tweet.

File: lib/embedded_tweet_view.dart [hunk 6]
```diff
@@ -66,10 +74,13 @@ class EmbeddedTweetView extends StatelessWidget {
     this.onTapImage,
     this.createdDateDisplayFormat,
     this.videoHighQuality = true,
+    this.showRepliesCount = false,
   }) : _tweetVM = TweetVM.fromApiV2Model(tweet, createdDateDisplayFormat);
 
   @override
   Widget build(BuildContext context) {
+    //final bool showReplies = showRepliesCount && _tweetVM.repliesCount != null;
```
Comment (wszeborowskimateusz) [inline lib/embedded_tweet_view.dart:82]: I guess this can be removed?

Comment (robertmrobo) [reply lib/embedded_tweet_view.dart:82]: > I guess this can be removed?

Done... ✅

**Comment by wszeborowskimateusz** [comment]
LGTM - I am merging it.

Thanks again for contribution.

**Pull request merged**